src/renderer/mod.rs

```diff
@@ -145,7 +147,7 @@ impl Renderer {
         self.cursor_renderer.prepare_frame()
     }
 
-    pub fn draw_frame(&mut self, root_canvas: &Canvas, dt: f32) {
+    pub fn draw_frame(&mut self, root_canvas: &Canvas, dt: f32, ime: &Ime, w: f32) {
```

Comment:
Can you give the  the `w` parameter a more descriptive name? We don’t use the shorthand for width anywhere else?